src/intrinsics/fb-www/fb-mocks.js

```diff
@@ -73,18 +68,12 @@ const fbMagicGlobalObjects = ["JSResource", "Bootloader"];
 
 function createMagicGlobalFunction(realm: Realm, global: ObjectValue | AbstractObjectValue, functionName: string) {
   global.$DefineOwnProperty(functionName, {
-    value: new NativeFunctionValue(realm, functionName, functionName, 0, (context, [string]) => {
-      invariant(string instanceof StringValue);
-      let value = `${functionName}("${string.value}")`;
-      let functionValue;
-
-      if (realm.fbLibraries.other.has(value)) {
-        functionValue = realm.fbLibraries.other.get(value);
-      } else {
-        functionValue = createAbstract(realm, "function", value);
-        realm.fbLibraries.other.set(value, functionValue);
-      }
-      invariant(functionValue instanceof AbstractValue);
+    value: new NativeFunctionValue(realm, functionName, functionName, 0, (context, args) => {
+      let functionValue = createAbstract(realm, "function");
```

Comment:
My impression is that AbstractValue constructor gives you more control. Maybe if you use that you won’t have to assign a private property.

Reply:
I tried that originally but I couldn’t get it to serialize right